lib/client/workloadendpoint.go

```diff
@@ -63,7 +64,16 @@ func (w *workloadEndpoints) Apply(a *api.WorkloadEndpoint) (*api.WorkloadEndpoin
 
 // Delete deletes an existing workload endpoint.
 func (w *workloadEndpoints) Delete(metadata api.WorkloadEndpointMetadata) error {
-	return w.c.delete(metadata, w)
+	// Validate the supplied Metadata.
+	if err := validator.Validate(metadata); err != nil {
+		return err
+	}
+
+	k, err := w.convertMetadataToKey(metadata)
+	if err != nil {
+		return err
+	}
+	return w.c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision})
```

Comment:
why did you change the if/else chain here?Couldnt we just add Revision in the call to backend? 
```go
// Untyped get interface for deleting a single API object.  This is called from the typed
// interface.
func (c *Client) delete(metadata unversioned.ResourceMetadata, helper conversionHelper) error {
	// Validate the supplied Metadata.
	if err := validator.Validate(metadata); err != nil {
		return err
	}

	if k, err := helper.convertMetadataToKey(metadata); err != nil {
		return err
	} else if err := c.Backend.Delete(&model.KVPair{Key: k, Revision: metadata.Revision}); err != nil {
		return err
	} else {
		return nil
	}
}
```